src/main/java/org/folio/entlinks/service/authority/AuthorityService.java

```diff
@@ -159,6 +161,20 @@ protected Authority deleteByIdInner(UUID id, boolean forced) {
     return repository.save(existed);
   }
 
+  private Authority validateOnUpdateAndGetExisting(UUID id, Authority modified) {
+    var existing = repository.findByIdAndDeletedFalse(id).orElseThrow(() -> new AuthorityNotFoundException(id));
+
+    var sourceFileId = Optional.ofNullable(modified.getAuthoritySourceFile())
+        .map(AuthoritySourceFile::getId)
+        .orElse(null);
+    if (sourceFileId != null && !sourceFileRepository.existsById(sourceFileId)) {
+      throw new AuthoritySourceFileNotFoundException(sourceFileId);
```

Comment:
with spring-boot upgrade to 3.4 there comes hibernate 6.6 and now it throws ```TransientObjectException``` when authority with non existing source file id set to ```sourceFileId``` field. Before, it was used to throw ```TransientPropertyValueException``` and we were handling this error to return 422 error response. Now, with ```TransientObjectException``` I don't see the point to handle it as it does not provide the context for which entity or property it represents the error other than just error message and thus decided to manually check the source file.